proto/storage/resource_collection.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "storage/user.proto";
+import "storage/policy.proto";
+
+package storage;
+
+message ResourceCollection {
+  message EmbeddedResourceCollection {
+    // 'id' is searchable to force a separate table
+    string id = 1 [(gogoproto.moretags) = 'search:"Embedded Collection ID",hidden sql:"fk(ResourceCollection:id)"'];
+  }
+
+  string id                                                = 1 [(gogoproto.moretags) = 'sql:"pk"']; // primary key
+  string name                                              = 2 [(gogoproto.moretags) = 'search:"Collection Name" sql:"unique"'];
+  string description                                       = 3;
+  google.protobuf.Timestamp created_at                     = 4;
+  google.protobuf.Timestamp last_updated                   = 5;
+  SlimUser created_by                                      = 6;
+  SlimUser updated_by                                      = 7;
+  // `resource_selectors` resolve as disjunction (OR) with each-other and with selectors from `embedded_collections`. For MVP, the size of resource_selectors will at most be 1 from UX standpoint.
+  repeated ResourceSelector resource_selectors             = 8;
+  repeated EmbeddedResourceCollection embedded_collections = 9;
+}
+
+message ResourceSelector {
+  // `rules` resolve as a conjunction (AND).
+  repeated SelectorRule rules = 1;
+}
+
+message SelectorRule {
+  // `field_name` can be one of the following:
+  // - Cluster
+  // - Cluster Label
+  // - Namespace
+  // - Namespace Label
+  // - Namespace Annotation
+  // - Deployment
+  // - Deployment Label
+  // - Deployment Annotation
+  string field_name         = 1;
```

Comment:
Any reason this isn't an enum?

Reply:
From direct messages with Mandar where I asked this of him:
Define the field_names as global string consts. It just keeps it the model flexible. TAL at features/list.go or pkg/booleanpolicy/fieldnames/list.go. The ListCollectionSelectors endpoint must serve these values.